index.js

```diff
@@ -32,19 +38,21 @@ export default function pLimit(concurrency) {
 	};
 
 	const enqueue = (function_, resolve, arguments_) => {
-		queue.enqueue(
-			AsyncResource.bind(run.bind(undefined, function_, resolve, arguments_)),
+		new Promise(r => {
```

Comment:
Oh, maybe add a comment saying that this is to preserve the AsyncResource context

Reply:
Thanks. I have added the comment.

- comment saying that this is to preserve the AsyncResource context 4aa57b7